pkg/integrations/v4/dm/id_provider.go

```diff
@@ -4,102 +4,151 @@
 package dm
 
 import (
+	"context"
 	"fmt"
 	"github.com/newrelic/infrastructure-agent/pkg/backend/identityapi"
 	"github.com/newrelic/infrastructure-agent/pkg/entity"
 	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/protocol"
+	"sync"
 )
 
-type RegisteredEntitiesNameToID map[string]entity.ID
-type UnregisteredEntitiesNamed map[string]UnregisteredEntity
+type registeredEntitiesNameToID map[string]entity.ID
+type unregisteredEntitiesNamed map[string]unregisteredEntity
 type reason string
 
-type UnregisteredEntity struct {
+type unregisteredEntity struct {
```

Comment:
I think 'unregisteredEntity' can cause some confusion. Can we use something else? for e.g. notRegisteredEntity?